mgmt/src/mgmt.c

```diff
@@ -71,6 +71,31 @@ mgmt_streamer_free_buf(struct mgmt_streamer *streamer, void *buf)
     streamer->cfg->free_buf(buf, streamer->cb_arg);
 }
 
+void
+mgmt_unregister_group(struct mgmt_group *group)
+{
+    struct mgmt_group *curr = mgmt_group_list, *prev;
+
+    if (curr && curr == group) {
+        mgmt_group_list = curr->mg_next;
+        return;
+    }
+
+    while (curr && curr != group) {
+        prev = curr;
```

Comment:
If `curr` is valid and equal to `group`, `prev` will remain uninitialized.

Reply:
Agreed, this needs a fix.

Reply:
curr will never be equal to group here because we take care of that condition above.